main.py

```diff
@@ -356,6 +362,53 @@ def __init__(self, *args):
 		self.actionExit.triggered.connect(self.onExitClicked)
 		self.actionSettings.triggered.connect(self.onSettingsClicked)
 
+		if not "search_for_updates" in self.settings.settings or not "buildId" in self.settings.settings:
+			self.settings.settings["search_for_updates"] = False
+			self.settings.settings["buildId"] = ""
+			self.settings.save()
+
+		if "search_for_updates" in self.settings.settings:
+			if self.settings.settings["search_for_updates"] == True:
+				if os.name == "nt":
+					projectJson = json.loads(requests.get("https://ci.appveyor.com/api/projects/mborgerson/xqemu-c5j6o/branch/master").text)
+					jobId = projectJson["build"]["jobs"][0]["jobId"]
+					print("Job ID: " + jobId)
+
+					if not jobId == self.settings.settings["buildId"]:
+						reply = QMessageBox.question(self, 'Update', "A new update is available, would you like to update?", QMessageBox.Yes, QMessageBox.No)
+						if reply == QMessageBox.Yes:
+							dir = os.path.dirname(os.path.abspath(self.settings.settings["xqemu_path"]))
+							print(dir)
+							if os.path.isdir(dir):
+								buildjobsJson = json.loads(requests.get("https://ci.appveyor.com/api/buildjobs/" + jobId + "/artifacts").text)
+
+								fileName = buildjobsJson[0]["fileName"]
+								print("Filename: " + fileName)
+
+								binaryUrl = "https://ci.appveyor.com/api/buildjobs/" + jobId + "/artifacts/" + fileName
+								print("Downloading " + fileName + " from " + binaryUrl)
+
+								with open(fileName, "wb") as file:
+									response = requests.get(binaryUrl)
+									file.write(response.content)
+								print("Downloaded")
+
+								self.settings.settings["buildId"] = jobId
+
+								print("Extracting...")
+								with zipfile.ZipFile(fileName, "r") as z:
+									z.extractall(dir)
```

Comment:
What happens, if we previously shipped files, which we don't ship anymore, which break newer versions?

* Old version: broken.dll + xqemu.exe
* New version: xqemu.exe

After the update, we'd still have the broken.dll in the xqemu.exe folder. So if xqemu.exe searches for a broken.dll, it will first find our old broken.dll, instead of a broken.dll that's in their Windows folder (or elsewhere), which might have been updated / fixed (which was the reason why we stopped including it).
So even after an upgrade, xqemu.exe might break now, because it accidentally loads an incompatible broken.dll.

It's not something that needs to be fixed for this PR, but we'll have to keep track of these kind of issues.
Generally, it's safer to have different folders for different versions (but that also comes with some problems).